pkg/util/lock.go

```diff
@@ -202,14 +200,22 @@ func (l *PgAdvisoryLock) getConn(connStr string, cur, maxRetries int) (*pgx.Conn
 		return nil, fmt.Errorf("max attempts reached. giving up on getting a db connection")
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), waitForConnectionTimeout)
-	defer cancel()
-
 	cfg, err := pgx.ParseConfig(connStr)
 	if err != nil {
 		return nil, fmt.Errorf("error parsing config connection: %w", err)
 	}
 
+	var (
+		ctx    = context.Background()
+		cancel context.CancelFunc
```

Comment:
nit: why declare this outside the if statement, it's only used inside. Just use the `:=` operator to create the cancel inside the if.  

The reason the current code is worse is that currently cancel survive to after line 217 but would be uninitiated if the if block isn't hit. Uninitiated variables are bad and should be avoided if possible.

Reply:
yep, that's my mistake. I thought it's being used somewhere else which is not local to that block. Anyways, thanks for pointing it out.

Reply:
Well, the reason is if i do not declare it here then the `:=` operator will make ctx too local in the block which is a compilation error. So, in order to allow ctx accessible to the below block and that goes into the ConnectConfig, I need to declare both of them outside.

> The reason the current code is worse is that currently cancel survive to after line 217 but would be uninitiated if the if block isn't hit.

If I am not wrong, I read somewhere that the difference between initializing `:=` vs `var` is that the memory is immediately used in `:=` but in `var`, its not used unless the variable is used somewhere. Do you think this is wrong? @cevian

Reply:
but still, it looks weird. I agree. I have a workaround that fits you suggestion. Will implement that and push. Thank you.